### ISSUE TITLE
Remove-Item doesn't support escape characters in wildcard patterns on Unix

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#3725]
Note: 
* Applies to Linux and macOS, but not to Windows
* Possibly related: #3724

Wildcard patterns- at least in the context of `Get-Item` - support `` ` `` as the escape character to represent literal `` ` `` instances in a path and to escape the following metacharacters: `* ? [ ]`

`Remove-Item` currently doesn't support that, neither with `-Path` nor with `-Include`.

Steps to reproduce
------------------

```powershell
# Create a file that contains a literal backtick and a literal substring that would be a valid
# sub-pattern if interpreted as a wildcard expression.
New-Item -Type File 'a`[b]'

# Match the newly created file using a wildcard expression with suitable escaping.
Remove-Item * -Include 'a```[b`]'  -WhatIf
Remove-Item -Path 'a```[b`]' -WhatIf
```

Expected behavior
-----------------

Both commands should find the target file and output information about how it _would_ be deleted.

Actual behavior
---------------

Both commands unexpectedly fail to find the file:
```
Remove-Item : An object at the specified path /Users/jdoe/a`[b] does not exist.
...
```

Curiously, the error message correctly quotes the correct _literal_ path that it claims doesn't exist.

Environment data
----------------

```powershell
PowerShell Core v6.0.0-alpha (v6.0.0-alpha.18) on macOS 10.12.4
PowerShell Core v6.0.0-alpha (v6.0.0-alpha.18) on Ubuntu 16.04
```

**Comment by iSazonov** [comment]
Is Move-Item affected too?

**Comment by mklement0** [comment]
@iSazonov: Good point; yes, `Move-Item` is affected too - I've retitled and revised the original post accordingly.
